lib/archethic_web/controllers/api/origin_public_key_payload.ex

```diff
@@ -0,0 +1,49 @@
+defmodule ArchEthicWeb.API.OriginPublicKeyPayload do
+  @moduledoc false
+
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  alias ArchEthic.Utils
+
+  alias ArchEthicWeb.API.Types.Hex
+  alias ArchEthicWeb.API.Types.PublicKey
+
+  embedded_schema do
+    field(:PublicKey, PublicKey)
+    field(:Certificate, Hex)
+  end
+
+  def changeset(params = %{}) do
+    %__MODULE__{}
+    |> cast(params, [
+      :PublicKey,
+      :Certificate
+    ])
+    |> validate_required([
+      :PublicKey
+    ])
+  end
+
+  def to_map(changes, acc \\ %{})
+
+  def to_map(%{changes: changes}, acc) do
+    Enum.reduce(changes, acc, fn {key, value}, acc ->
+      key = Macro.underscore(Atom.to_string(key))
```

Comment:
Why do you want to convert key as string ?

Reply:
Thought to check the post request certificate and the correct certificate is matched so, would need string but didn't used.

Now Deleted as they is no usage